starter-code/main.js

```diff
@@ -53,6 +53,195 @@ mongoClient.connect(`mongodb://localhost:27017/crunchbase`, (error, db) => {
             })
             break;
 
+            case "4":
+              //4.- List by name all companies founded in february of 2004.
+            db.collection('companies').find({ $and: [ { founded_year: 2004 }, { founded_month: 2 } ] }).count((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "5":
+              //5.- List by name all companies founded in the summer of 2004 (april to june) sorted by date.
+            db.collection('companies').find({ $and: [ {founded_year: 2004}, { founded_month: {$gt: 3}}, {founded_month: {$lt: 7}}]},{founded_month: 1}).count((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+            case "6":
+              //6.- What companies have offices in "Barcelona".
+            db.collection('companies').find({ "offices.city": { $eq: 'Barcelona' } }).count((error, result) => {
```

Comment:
Una equivalencia más sencilla:
```suggestion
            db.collection('companies').find({ "offices.city": 'Barcelona'  }).count((error, result) => {
```